src/aiq/tool/mcp/mcp_client.py

```diff
@@ -83,49 +87,122 @@ def _generate_field(field_name: str, field_properties: dict[str, Any]) -> tuple:
     return create_model(f"{_generate_valid_classname(name)}InputSchema", **schema_dict)
 
 
-class MCPSSEClient:
+class MCPBaseClient(ABC):
+    """
+    Base client for creating a session and connecting to an MCP server
+    """
+
+    def __init__(self, url: str):
+        self.url = url
+
+    @abstractmethod
+    @asynccontextmanager
+    async def connect_to_server(self):
+        """
+        Establish a session with an MCP server within an async context
+        """
+        pass
+
+
+class MCPSSEClient(MCPBaseClient):
     """
     Client for creating a session and connecting to an MCP server using SSE
 
     Args:
       url (str): The url of the MCP server
     """
 
-    def __init__(self, url: str):
-        self.url = url
-
     @asynccontextmanager
-    async def connect_to_sse_server(self):
+    async def connect_to_server(self):
         """
-        Establish a session with an MCP SSE server within an aync context
+        Establish a session with an MCP SSE server within an async context
         """
         async with sse_client(url=self.url) as (read, write):
             async with ClientSession(read, write) as session:
                 await session.initialize()
                 yield session
 
 
-class MCPBuilder(MCPSSEClient):
+class MCPStdioClient(MCPBaseClient):
+    """
+    Client for creating a session and connecting to an MCP server using stdio
+
+    Args:
+      url (str): The command to run
+      args (list[str] | None): Additional arguments for the command
+      env (dict[str, str] | None): Environment variables to set for the process
+    """
+
+    def __init__(self, url: str, args: list[str] | None = None, env: dict[str, str] | None = None):
+        super().__init__(url)
+        self._args = args
+        self._env = env
+
+    @asynccontextmanager
+    async def connect_to_server(self):
+        """
+        Establish a session with an MCP server via stdio within an async context
+        """
+        server_params = StdioServerParameters(command=self.url, args=self._args or [], env=self._env)
+        async with stdio_client(server_params) as (read, write):
+            async with ClientSession(read, write) as session:
+                await session.initialize()
+                yield session
+
+
+class MCPBuilder(MCPBaseClient):
```

Comment:
We should get rid of `MCPBuilder` and just use `MCPBaseClient` in its place. It's current purpose is just to create clients based on the client type and then forward arguments to the client. The rest of the logic in this class can be put into `MCPBaseClient`

Reply:
Agree, good suggestion